tests/tape/test_qnode.py

```diff
@@ -330,6 +330,12 @@ def func(x):
             == "<QNode: wires=1, device='default.qubit.autograd', interface='autograd', diff_method='best'>"
         )
 
+     def test_diff_method_None(self):
+        """Test if diff_method=None works as intended"""
+        dev = qml.device("default.qubit", wires=1)
+        qn = qml.Qnode(dummyfunc, dev, diff_method=None)
+        assert qn.interface is None

```

Comment:
For testing, it would also be useful to both run and try to differentiate the QNode to ensure that the correct behaviour occurs; presumably it will throw an error if you try to do something like this?

```
g = qml.grad(qn)
g()
```

Reply:
Depending on the type of circuit, it may or may not result in an error. In my trials, if the output depends on input, it will give some error that is not always the same (depends on the type of input). If the output is not depending on the input, there is no error.
It's non-sensical to try to get a gradient when explicitly specifying `diff_method=None`.
Thus, I think it best to leave the test as is.

Reply:
Hi @ankit27kh, I think Olivia was more so proposing to add new test cases in addition to this one. It would be definitely great to have more, as the effect of allowing `diff_method=None` doesn't seem to be tested at the moment.

The cases that you mention each would make great test cases:

1. Testing if the output depends on the input (with the various variants)
2. Testing if the output doesn't depend on the input

Also, if it turns out that it makes more sense, we could raise a `qml.QuantumFunctionError` in general whenever a user wants to differentiate a QNode that has `diff_method=None`.

Reply:
@antalszava The latter option seems best. Currently, using gradient raises errors that are not helpful for the user at all. For example:
```
TypeError: must be real number, not ArrayBox
``` 
> we could raise a `qml.QuantumFunctionError` in general whenever a user wants to differentiate a QNode that has `diff_method=None`.

This should help and stop any execution before it reaches this place.

Regarding, adding more tests here, as I explain in another comment below, this is only a simple patch.

If anything, it is only transferring workflow to `interface=None` when the QNode is initialized. These cases should have been checked when `interface=None` was added. Even, the new `qml.QuantumFunctionError`, if implemented, should be added to `interface=None` part of the code and not here.

Reply:
Hi @ankit27kh, thanks for the comments! Although `diff_method = None` will greatly leverage `interface=None`, checking that `interface=None` was set correctly is only a test for the unit itself. Unit testing might not be enough to test the entire functionality of `diff_method=None`, as there could be unexpected sharp edges along the entire pipeline. The test cases listed above would constitute integration tests, to check that the entire pipeline works when setting `diff_method=None`. So it would definitely be definitely nice to have some of those here.